linera-core/src/client/mod.rs

```diff
@@ -2565,9 +2568,12 @@ where
         &self,
         bytes: Vec<Vec<u8>>,
     ) -> Result<ClientOutcome<Certificate>, ChainClientError> {
-        let blobs = bytes.into_iter().map(Blob::new_data);
+        let mut blobs = Vec::new();
+        for byte_vec in bytes {
+            blobs.push(Blob::new_data(byte_vec));
+        }
```

Comment:
```suggestion
        let blobs = bytes.into_iter().map(Blob::new_data).collect::<Vec<_>>();
```
would allocate the right vector size right away.